R/format_PFN.R

```diff
@@ -0,0 +1,589 @@
+                  ReleaseAlive = ifelse(.data$RTYPE == "X", FALSE, TRUE),
+                  ExperimentID = dplyr::case_when(.data$COND == "M" ~ "SURVIVAL",
+                                                  is.na(.data$COND) ~ NA_character_)
+    )
+
+
+
+  ## 6) Add chick measurement and brood ID data from brood table (primary data)
+
+  # Extract chick mass, tarsus, and BroodID from brood table
+  Chick_Capture_list <- list()
+  for(i in 1:nrow(Primary_data)){
+    Chick_Capture_list[[i]] <- data.frame(IndvID = unname(t(Primary_data[i, paste0("Young", c(1:11))])),
+                                          BroodID = Primary_data$BroodID[i], # Will be removed later
+                                          #CaptureDate = as.Date(Primary_data$YoungDate[i], format = "%d/%m/%Y"),
+
+                                          Mass_B = as.numeric(Primary_data[i, paste0("Weight.Y", c(1:11))]),
+                                          Tarsus_B = as.numeric(Primary_data[i, paste0("Tarsus.Y", c(1:11))]),
+                                          stringsAsFactors = FALSE)
+  }
+
+  Chick_Capture_data <- dplyr::bind_rows(Chick_Capture_list)
+  Chick_Capture_data <- subset(Chick_Capture_data, !is.na(IndvID))
+
+  # Merge additional chick data into capture data
+  #Capture_data <- Capture_data %>%
+  #  dplyr::left_join(Chick_Capture_data, by = c("IndvID", "CaptureDate")) %>%
+
+  Capture_data <- Capture_data %>%
+    dplyr::left_join(Chick_Capture_data, by = "IndvID") %>%
+
+  # Summarise information on chick mass and tarsus measurements
+    #dplyr::mutate(Mass = dplyr::case_when(!is.na(.data$Mass_CMR) ~ .data$Mass_CMR,
+    #                                      is.na(.data$Mass_CMR) & !is.na(.data$Mass_B) ~ .data$Mass_B),
+    #              Tarsus = dplyr::case_when(!is.na(.data$Tarsus_CMR) ~ .data$Tarsus_CMR,
+    #                               is.na(.data$Tarsus_CMR) & !is.na(.data$Tarsus_B) ~ .data$Tarsus_B)) %>%
+    dplyr::mutate(Mass = dplyr::case_when(!is.na(.data$Mass_CMR) ~ .data$Mass_CMR,
+                                          is.na(.data$Mass_CMR) & !is.na(.data$Mass_B) & .data$Age_observed == 1 ~ .data$Mass_B,
+                                          is.na(.data$Mass_CMR) & !is.na(.data$Mass_B) & .data$Age_observed != 1 ~ NA_real_),
+                  Tarsus = dplyr::case_when(!is.na(.data$Tarsus_CMR) ~ .data$Tarsus_CMR,
+                                            is.na(.data$Tarsus_CMR) & !is.na(.data$Tarsus_B) & .data$Age_observed == 1 ~ .data$Tarsus_B,
+                                            is.na(.data$Tarsus_CMR) & !is.na(.data$Tarsus_B) & .data$Age_observed != 1 ~ NA_real_)) %>%
+
+    ## 7) Exclude entries not included in the standard format
+
+    # Remove all individuals from species not included in Species_codes
+    dplyr::filter(.data$Species %in% Species_codes$Code) %>%
+
+    # Remove all observations not involving a capture/dead recovery (i.e. resightings)
+    dplyr::filter(.data$RTYPE != 'O') %>%
```

Comment:
I disagree with you here. It is true that these do contain information about an individual being alive, but for any mark-recapture modelling we need to be able to distinguish between resightings and recaptures, because the probability of resighting an alive individual and the probability of recapturing an alive individual are not the same. If resightings are treated like recaptures in a model, estimates will get biased because we ignore that there are two different data-generating processes. 

At the moment, the SPI-birds format distinguishes between live recaptures and dead recoveries using the columns CaptureAlive and ReleaseAlive. Information in these columns can not be used to distinguish recaptures and resightings. 
Therefore, I want to remove resightings and limit Capture_data to actual captures, unless we want to introduce a new column which specifies the type of capture (e.g. "marking", "recapture", "resighting", "recovery").

Reply:
I would strongly prefer the latter option. We should avoid excluding data whenever possible, otherwise we (the developers) are making a subjective decision about what data are useful and what the SPI-Birds data should/shouldn't be used for. Not everybody will be building mark-recapture models and in other cases having resightings data might be useful. Plus, it's always much easier to remove unwanted data than to add back in data.

Reply:
We agreed that we should implement the latter option at a later point (i.e. next time we update the standard protocol). I made an issue for this (#119 ).